src/js/utils/CreateEvent.js

```diff
@@ -0,0 +1,14 @@
+export function createEvent(event_type) {
```

Comment:
While this `toggle_event` variable recycle will work (and is sometimes preferred), I think a more direct approach would be more readable; try to return the `CustomEvent` immediately, when that fails instantiate the new `const toggle_event`.

I also think the explicit variables (`can_bubble`, etc.) aren't really necessary; the method signature of `initCustomEvent` should explain what the arguments mean. Maybe also add some documentation why you are trying the other deprecated approach (`initCustomEvent`, I assume IE support?)

Good idea to move this to a util!